presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/thrift/TestDynamicMetastoreConfig.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore.thrift;
+
+import org.apache.thrift.TException;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestDynamicMetastoreConfig
+{
+    @Test
+    public void testMetastoreDiscoveryUri() throws TException
```

Comment:
This is not usually the way we test configs. Check other examples in the code. For example `TestMetastoreConfig`

Reply:
Will do.

Reply:
Added unit test using TestStaticMetastoreConfig as reference. However, assertFullMapping fails, probably because of the parameter being deserialised using JsonCreator.